examples/BlazorServerApp/README.md

```diff
@@ -0,0 +1,69 @@
+
+This app demonstrates how to use the Feature Management library in Blazor apps.
+
+This app uses two feature flags: "BrowserEnhancement" and "Beta".
+
+``` json
+"FeatureManagement": {
+    "BrowserEnhancement": {
+        "EnabledFor": [
+            {
+                "Name": "Browser",
+                "Parameters": {
+                    "AllowedBrowsers": [ "Edge" ]
+                }
+            }
+        ]
+    },
+    "Beta": {
+        "EnabledFor": [
+            {
+                "Name": "Targeting",
+                "Parameters": {
+                    "Audience": {
+                        "DefaultRolloutPercentage": 50,
+                        "Exclusion": {
+                            "Groups": [
+                                "Guests"
+                            ]
+                        }
+                    }
+                }
+            }
+        ]
+    }
+}
+```
+
+The `"BrowserEnhancement"` feature is enabled when the user is using the allowed browsers, in this example, the Edge browser. If the `"BrowserEnhancement"` feature is on, the color of the top bar will be dark blue. Otherwise, it will be white.
+
+The `"Beta"` feature uses the [`Targeting`](https://github.com/microsoft/FeatureManagement-Dotnet?tab=readme-ov-file#targeting) filter to evaluate when to activate. The `"Guests"` group, consisting of the unauthenticated users, will be excluded from the audience. Other users will have 50% chance to fall into the default rollout. When the `"Beta"` feature is enabled, the **BETA** content will be displayed.
+
+## HttpContext
+
+In regular ASP.NET Core web app, the most common way to get targeting context is through the `HttpContextAccessor`. However, `IHttpContextAccessor` must be avoided with interactive rendering of Razor component in the Blazor server app because there isn't a valid `HttpContext` available.
+
+[The recommended approach](https://learn.microsoft.com/en-us/aspnet/core/blazor/security/server/interactive-server-side-rendering?view=aspnetcore-7.0#ihttpcontextaccessorhttpcontext-in-razor-components) to passing http context in Blazor apps is to copy the data into a scoped service. This app obtains the `"User-Agent"` information from the `HttpContext` and passes it to a scoped service called `UserAgentContextProvider`. `UserAgentContextProvider` will be consumed by the `BrowserFilter`.
+
+## User Authentication State
+
+This app uses [cookie authentication](https://learn.microsoft.com/en-us/aspnet/core/security/authentication/cookie?view=aspnetcore-6.0). When the app is loaded, it has 2/3 chance to be an authenticated user. The details can be found [here](./examples/BlazorServerApp/Pages/_Host.cshtml).
+
+Rather than `HttpContext`, the [`AuthenticationStateProvider`](https://learn.microsoft.com/en-us/aspnet/core/blazor/security/?view=aspnetcore-8.0#authenticationstateprovider-service) service is used to obtain the user authentication state information for setting targeting context.
```

Comment:
I would recommend calling out `MyTargetingContextAccessor` and providing a link to it for more details.